lib/src/hex_util.rs

```diff
@@ -12,29 +12,79 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#![allow(missing_docs)]
+//! Hex string helpers.
 
 const REVERSE_HEX_CHARS: &[u8; 16] = b"zyxwvutsrqponmlk";
 
-fn to_forward_hex_digit(b: u8) -> Option<u8> {
-    let value = match b {
-        b'k'..=b'z' => b'z' - b,
-        b'K'..=b'Z' => b'Z' - b,
-        _ => return None,
-    };
-    if value < 10 {
-        Some(b'0' + value)
-    } else {
-        Some(b'a' + value - 10)
+fn reverse_hex_value(b: u8) -> Option<u8> {
+    match b {
+        b'k'..=b'z' => Some(b'z' - b),
+        b'K'..=b'Z' => Some(b'Z' - b),
+        _ => None,
     }
 }
 
-pub fn to_forward_hex(reverse_hex: impl AsRef<[u8]>) -> Option<String> {
-    reverse_hex
-        .as_ref()
-        .iter()
-        .map(|b| to_forward_hex_digit(*b).map(char::from))
-        .collect()
+/// Decoded hex or error, which may include odd-length byte.
+#[derive(Clone, Debug)]
+pub enum DecodeHexResult {
```

Comment:
This type seems very similar to `HexPrefix` (well `Result<HexPrefix, ...>`). Do we need both?

Reply:
They're similar, but it depends on caller whether odd-length bytes is acceptable or not, so I think the intermediate `Result`-ish type helps clarify that.